### DISCUSSION TITLE
QT5 app

### DISCUSSION THREAD

**Discussion by floringhimie** [pywinauto/pywinauto#1352] [General]
Hi, first of all great job on this project.
I'm trying to make a POC for a testing project and this library is one candidate.
Internally we are using a dedicated platform for testing a QT5 application(Squish from Froglogic now Qt).
I did some comparison between Squish and pywinauto and there are some difference between how elements are identified.
I don't understand why we are not able to see more information in the app as Squish sees, the same is happening with Inspect.exe
Basically the pywinauto see what we see in Inspect.exe but Squish see more.
It identify elements and buttons with more information as text color, font name, images are identified with path ../...svg and some buttons that pywinauto that are identified with no name or auto_id in Squish we have a full name. If a button has a icon (for example close button has a X icon, in Squish we see also the button icon as child).
Any idea why?
Thank you!

**Comment by junkmd** [comment]
Hi,

I am not familiar with how Squish identifies and parses Qt gui elements.

However, `pywinauto` does not rely on a specific GUI framework, but uses accessibility (e.g. UIAutomation) and OS APIs (e.g. win32api) to retrieve gui element information.

So, I think that `pywinauto` cannot retrieve detailed information as compared to framework-specific tools such as Squish.

**Comment by vasily-v-ryabov** [comment]
Squish most probably use DLL injection approach for QML apps. We can create similar "qml" backend for pywinauto, but it requires pretty much work. Similar approach is already implemented for WPF apps (in pull requests at the moment). If somebody can help with it, I can advice where to dig.

**Comment by vasily-v-ryabov** [comment]
I'm planning to tidy up current activities during New Year holidays and post the DLL injection package to PyPI ("injected" repo in the organization). Then I can merge WPF backend with DLL injection, and we can continue in similar directions.

**Comment by vasily-v-ryabov** [comment]
See also #1353. Maybe it is possible to join efforts.

**Comment by floringhimie** [comment]
Yes, I think it is the same issue here.
